src/terminal/parser/InputStateMachineEngine.cpp

```diff
@@ -790,6 +858,113 @@ DWORD InputStateMachineEngine::_GetModifier(const size_t modifierParam) noexcept
     return modifierState;
 }
 
+// Method Description:
+// - Synthesize the button state for the Mouse Input Record from an SGR VT Sequence
+// - Here, we refer to and maintain the global state of our mouse.
+// - Mouse wheel events are added at the end to keep them out of the global state
+// Arguments:
+// - wch: the wchar_t representing whether the button was pressed or released
+// - parameters: the wchar_t to get the mapped vkey of. Represents the direction of the button (down vs up)
+// - buttonState: Recieves the button state for the record
+// - eventFlags: Recieves the special mouse events for the record
+// Return Value:
+// true iff we were able to synthesize pdwButtonState
+bool InputStateMachineEngine::_UpdateSGRMouseButtonState(const wchar_t wch,
+                                                         const std::basic_string_view<size_t> parameters,
+                                                         DWORD& buttonState,
+                                                         DWORD& eventFlags)
+{
+    if (parameters.empty())
+    {
+        return false;
+    }
+
+    // Starting with the state from the last mouse event we received
+    buttonState = _mouseButtonState;
+    eventFlags = 0;
+    bool success = true;
+
+    // The first parameter of mouse events is encoded as the following two bytes:
+    // BBDM'MMBB
+    // Where each of the bits mean the following
+    //   BB__'__BB - which button was pressed/released
+    //   MMM - Control, Alt, Shift state (respectively)
+    //   D - flag signifying a drag event
+    // This retrieves the 2 MSBs and concatenates them to the 2 LSBs to create BBBB in binary
+    // This represents which button had a change in state
+    const auto buttonID = (til::at(parameters, 0) & 0x3) | ((til::at(parameters, 0) & 0xC0) >> 3);

```

Comment:
Wait am I crazy? Shouldn't this be `>>4`? I thought bit shifting worked like this:

```
(param & 0x3) | (param  & 0xC0) >> 0); // BB__'__BB
(param & 0x3) | (param  & 0xC0) >> 1); // _BB_'__BB
(param & 0x3) | (param  & 0xC0) >> 2); // __BB'__BB
(param & 0x3) | (param  & 0xC0) >> 3); // ___B'B_BB
(param & 0x3) | (param  & 0xC0) >> 4); // ____'BBBB
```

Reply:
If a test didn't catch this.... should we have one that covers this?

Reply:
Funny enough, the test that covers this is the scrolling one because scrolling touches the 2nd MSB hahaha